apluslms_roman/observer.py

```diff
@@ -151,38 +153,81 @@ def container_msg(self, step, msg):
         self._send_message(Message.CONTAINER_MSG, step, msg)
 
     def result_msg(self, result):
+        state = (StepState.SUCCEEDED if result.ok
+            else (StepState.CANCELLED if result.cancelled
+                else StepState.FAILED))
+        self._state_update(result.step, state)
         self._send_message(Message.RESULT_MSG, result.step, (result.code, result.error))
 
 
-ENTER_STATE_TEXTS = {
-    StepState.PREFLIGHT: "Pre-Flight tasks..",
-    StepState.PENDING: "Pending..",
-    StepState.RUNNING: "Running..",
-    StepState.POSTFLIGHT: "Post-Flight tasks..",
-    StepState.FAILED: "Failed!",
-    StepState.CANCELLED: "\rCancelled..",
-}
 class StreamObserver(BuildObserver):
-    def __init__(self, stream=None):
+    def __init__(self, stream=None, colors=True):
         super().__init__()
         self.stream = stream or sys.stdout
+        self.colors = colors
+        self.start_time = -1
```

Comment:
All of these should probably be prefixed with `_` too (including the stream, which apparently was forgotten).